meta-facebook/wc-mb/src/shell/commands/plat_info_shell.c

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "plat_info_shell.h"
+#include "plat_class.h"
+
+int cmd_wcmb_info_print(const struct shell *shell, size_t argc, char **argv)
+{
+	;
+	shell_print(shell, "========================{PLATFORM INFO}========================");
+	shell_print(shell, "* NAME:          Waimea Canyon platform command");
+	shell_print(shell, "* DESCRIPTION:   Commands that could be used to debug or validate.");
+	shell_print(shell, "* Note:          none");
+	shell_print(shell, "------------------------------------------------------------------");
+	shell_print(shell, "* SYSTEM CLASS:  %s",
+		    get_system_class() == SYS_DUAL ? "Dual" : "Single");
+	shell_print(shell, "* SYSTEM SOURCE: %s",
+		    get_source_class() == SRC_MAIN ? "Main" : "Sencond");
```

Comment:
Refactor the cmd_info_print function to something like this

```
void print_main_info(const struct shell *shell)
{
        shell_print(
                shell,
                "========================{SHELL COMMAND INFO}========================================");
        shell_print(shell, "* NAME:          Platform command");
        shell_print(shell, "* DESCRIPTION:   Commands that could be used to debug or validate.");
        shell_print(shell, "* DATE/VERSION:  none");
        shell_print(shell, "* CHIP/OS:       %s - %s", CONFIG_BOARD, RTOS_TYPE);
        shell_print(shell, "* Note:          none");
        shell_print(shell, "------------------------------------------------------------------");
        shell_print(shell, "* PLATFORM:      %s-%s", PLATFORM_NAME, PROJECT_NAME);
        shell_print(shell, "* BOARD ID:      %d", BOARD_ID);
        shell_print(shell, "* STAGE:         %d", PROJECT_STAGE);
        shell_print(shell, "* FW VERSION:    %d.%d", FIRMWARE_REVISION_1, FIRMWARE_REVISION_2);
        shell_print(shell, "* FW DATE:       %x%x.%x.%x", BIC_FW_YEAR_MSB, BIC_FW_YEAR_LSB,
                ⦙   BIC_FW_WEEK, BIC_FW_VER);
        shell_print(shell, "* FW IMAGE:      %s.bin", CONFIG_KERNEL_BIN_NAME);
        shell_print(
                shell,
                "========================{SHELL COMMAND INFO}========================================");
}

__weak int cmd_info_print(const struct shell *shell, size_t argc, char **argv)
{
        print_main_info(shell);
        return 0;
}
```

Then we can override this function and call the same print_main_info and then print the additional system_class info.